bug/232 fixed landing page not applying the correct field

brianandres2 opened the pull request:
ended up being half a line of code - setting the 'fields' variable to the current active field instead of always 'all' when the fields variable is not defined. Weird case that only applies whenever going from landing page to search page.

brianandres2 committed:
fixed

paulpilone commented:
@brianandres2 Curious why that value isn't set if I hit "Enter" from the landing page but it _is_ set if I click the Search button is clicked. I ask because while your change seems to work it feels like there shouldn't be any difference in behavior between those two actions.

brianandres2 commented:
Short Answer: they called separate functions that were very similar, but when I changed code to store fields in the URL on the search page, it messed with the function that was called when pushing enter or the magnifying glass, but left the other function that was called when pushing the button still working through a coincidence.

Long Answer: Originally there was no variable that stored multiple fields, and only a variable that stored the current active field. It was called "active_fields", and seeing as it had an 's' at the end, I stole that name for the variable that would store all the fields in the url, and renamed the current one (that stored the current active field) to 'active_field'. This function is called when reloading the page or being sent there from the home page when pushing enter or the magnifying glass (while the button would call a separate function, I'll get more to that later).

When you reload the page, the URL stores all the information so it has no problem loading. However, when you come from the landing page, if the variable that stores all the fields was empty (which it will always be when coming from the landing page), it would be by default set to 'all' (left over code from setting the 'active_fields' variable to 'all' by default, back when it represented the current active field (which was redundant but that would require more explaining)) This made it so that the 'fields' variable in the URL would always be set to 'all' if you clicked enter or the magnifying glass from the landing page.

But if you clicked the button on the landing page that leads to the unchanged other function, the active_fields variable was still set to the current active field (since originally the active_fields variable _was_ the current active field). This is actually what we want from the landing page, since at this point there are no fields to be stored in the variable that remembers all the fields, and we want the first field to be the current active field. So by coincidence, the button still works while the enter key and magnifying glass did not.

brianandres2 commented:
@paulpilone  (forgot to ping you in last comment)

this is what the button function is: 

```
  handleSearchButtonClick() {
    const route = formSearchRoute({
      active_fields: activeFieldsString(this.props.fields),
      active_options: activeAdvancedOptionsString(this.props.options),
      active_search: this.props.searchReducer.search,
    });

    this.props.history.push(route);
  }
```

this is what the magnifying glass/enter key function is: (its bigger because it has more information from the URL):

```
navigateToSearch(activeSearch, fields = activeFieldsString(this.props.fields)) {

    const active_sort = activeSortOption(this.props.options);
    const route = formSearchRoute({
      active_field: activeFieldsString(this.props.fields),
      active_fields: fields,
      active_tags: activeTagsString(this.props.searchReducer.activeFilters),
      active_options: activeAdvancedOptionsString(this.props.options),
      active_sort: active_sort.filter,
      active_search: activeSearch,
    });

    if ( window.location.search !== route.search ) {
      this.props.history.push(route);
    }

  }
```

paulpilone reviewed (approved):
Looks good! Sorry for the delay!
